pkg/cluster/px/px.go

```diff
@@ -604,6 +621,29 @@ func (ops *pxClusterOps) upgradePX(newVersion string) error {
 	return nil
 }
 
+// setEnv set the envVar on a given key with a given value
+func setEnv(envVars []corev1.EnvVar, key, value string) []corev1.EnvVar {
```

Comment:
this could be written simpler with a for loop. Remove the index based logic.

Iterate on `envVars` and check if `key` exists. If yes, update. After loop is done, if not found, append.